### DISCUSSION TITLE
GImpact collision algorithm is not dispatched in contactPairTest

### DISCUSSION THREAD

**Discussion by i-to** [bulletphysics/bullet3#3747] [General]
Reproduction sequence:
1. Setup `btCollisionWorld`.
2. Register GImpact algorithm via `btGImpactCollisionAlgorithm::registerAlgorithm`.
3. Create two collision objects containing `btGImpactMeshShape` shapes, add them to world.
4. Invoke `btCollisionWorld::contactPairTest` for two objects.

Result: `btEmptyAlgorithm` stub is dispatched for testing and actual collisions are never reported.
This behavior appeared in commit 9ee1c4ec245ff0fc860fc56c78910d4ab0dd0579. In earlier versions, GImpact algorithm would be used.

Would it make sense to rename `btCollisionWorld::contactPairTest` to `closestPointPairTest` 
and then use `BT_CONTACT_POINT_ALGORITHMS` in `contactPairTest` for algorithm lookup?

**Comment by jrz371** [comment]
As a workaround you can update your collision objects, call performDiscreteCollisionDetection on the collision world, then iterate over the manifolds in the dispatcher. This should at least give you the collisions.
